src/metatensor/models/utils/compute_loss.py

```diff
@@ -1,33 +1,59 @@
-from typing import Dict, List
+import warnings
+from typing import Dict, List, Union
 
 import torch
 from metatensor.torch import Labels, TensorBlock, TensorMap
-from metatensor.torch.atomistic import System
+from metatensor.torch.atomistic import (
+    ModelEvaluationOptions,
+    System,
+    register_autograd_neighbors,
+)
 
+from .export import is_exported
 from .loss import TensorMapDictLoss
+from .neighbor_list import calculate_neighbor_lists
 from .output_gradient import compute_gradient
 
 
+# Ignore metatensor-torch warning due to the fact that positions/cell
+# already require grad when registering the NL
+warnings.filterwarnings(
+    "ignore",
+    category=UserWarning,
+    message="neighbors",
+)  # TODO: this is not filtering out the warning for some reason
+
+
 def compute_model_loss(
     loss: TensorMapDictLoss,
-    model: torch.nn.Module,
+    model: Union[torch.nn.Module, torch.jit._script.RecursiveScriptModule],
     systems: List[System],
     targets: Dict[str, TensorMap],
 ):
     """
     Compute the loss of a model on a set of targets.
 
     :param loss: The loss function to use.
-    :param model: The model to use.
+    :param model: The model to use. This can either be a model in training
+        (``torch.nn.Module``) or an exported model
+        (``torch.jit._script.RecursiveScriptModule``).
     :param systems: The systems to use.
     :param targets: The targets to use.
 
     :returns: The loss as a scalar `torch.Tensor`.
     """
     # Assert that all targets are within the model's capabilities:
-    if not set(targets.keys()).issubset(model.capabilities.outputs.keys()):
+    if not set(targets.keys()).issubset(_get_capabilities(model).outputs.keys()):
         raise ValueError("Not all targets are within the model's capabilities.")
 
+    # calculate neighbor lists (only for an exported model)
+    if is_exported(model):
+        requested_neighbor_lists = model.requested_neighbors_lists()
+        systems = [
+            calculate_neighbor_lists(system, requested_neighbor_lists)
```

Comment:
Suggestion: I'm sorry for being picky, but for a function with a name `calculate_neighbor_lists()` I would rather expect to get a `TensorBlock` with a computed NL, rather than the new system with the NL attached. 

Here I would rather re-name the function to make it consistent with its functionality, or rather split this by explicit NLs calculation and NLs adding parts.

Reply:
You're right, I'll rename it

Reply:
Cool, thank you